launch_frontend/launch_frontend/parser.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for Parser methods."""
+
+import launch
+
+import launch_ros
+
+from .entity import Entity
+
+
+def get_attr_or_default(entity, attr, default_value):
+    """Return the attr value of entity or default_value."""
+    try:
+        return entity.__getattr__(attr)
+    except AttributeError:
+        return default_value
+
+
+def get_attr_or_none(entity, attr):
+    """Return the attr value of entity or None."""
+    return get_attr_or_default(entity, attr, None)
+
+
+def str_to_bool(string):
+    """Convert xs::boolean to python bool."""
+    if not string:
+        return None
+    if string == 'true':
+        return True
+    if string == 'false':
+        return False
+    raise RuntimeError('Expected "true" or "false", got {}'.format(string))
+
+
+def get_dictionary_from_key_value_pairs(pairs):
+    """Get dictionary from key-value pairs."""
+    if not pairs:
+        return None
+    return {pair.name: pair.value for pair in pairs}
+
+
+def evaluate_predicate(predicate):
+    """Evaluate a predicate to true or false."""
+    # TODO(ivanpauno): Implement me please
+    return True
+
+
+def filtrate_with_predicates(items):
```

Comment:
@ivanpauno conditions should be dealt with [`launch.Condition`](https://github.com/ros2/launch/blob/master/launch/launch/condition.py#L24) instances.